alonzo/impl/src/Cardano/Ledger/Alonzo/TxInfo.hs

```diff
@@ -395,81 +488,102 @@ runPLCScript proxy (CostModel cost) scriptbytestring units ds =
     (transExUnits units)
     scriptbytestring
     ds of
-    (_, Left e) -> explain_plutus_failure proxy scriptbytestring e ds
+    (_, Left e) -> explain_plutus_failure proxy scriptbytestring e ds (CostModel cost) units
     (_, Right ()) -> Passes
 
 -- | Explin why a script might fail. Scripts come in two flavors. 1) with 3  data arguments [data,redeemer,context]
 --   and  2) with 2 data arguments [redeemer,context]. It pays to decode the context data into a real context
 --   because that provides way more information. But there is no guarantee the context data really can be decoded.
-explain_plutus_failure :: forall era. Show (Script era) => Proxy era -> SBS.ShortByteString -> P.EvaluationError -> [P.Data] -> ScriptResult
-explain_plutus_failure _proxy scriptbytestring e [dat, redeemer, info] =
+explain_plutus_failure ::
+  forall era.
+  Show (Script era) =>
+  Proxy era ->
+  SBS.ShortByteString ->
+  P.EvaluationError ->
+  [P.Data] ->
+  CostModel ->
+  ExUnits ->
+  ScriptResult
+explain_plutus_failure _proxy scriptbytestring e ds@[dat, redeemer, info] cm eu =
   -- A three data argument script.
   let ss :: Script era
       ss = PlutusScript scriptbytestring
       name :: String
       name = show ss
    in case P.fromData info of
-        Nothing -> Fails [line]
+        Nothing -> Fails [PlutusFailure line db]
```

Comment:
also I forget, but could we put the structured type in here and just print it specially later?

Reply:
I think the train has left the station, sorry. But as for you question, I don't know, we would have to write our own show for it to produce the base64 in the logs messages, which is maybe a weird/not-always-what-you-want show instance?

Reply:
You're using `Show` for log messages? I'm not totally sure that's a good idea, but if that's the situation then okay.

Reply:
I'm not actually sure, to be honest.